src/IB/IBExplicitHierarchyIntegrator.cpp

```diff
@@ -381,7 +387,14 @@ IBExplicitHierarchyIntegrator::postprocessIntegrateHierarchy(const double curren
                                          getGhostfillRefineSchedules(d_object_name + "::u"),
                                          new_time);
 
-    // This class is not responsible for postprocessing any other objects so proceed to the base class:
+    // Update the marker points, should they exist:
+    if (d_marker_points)
+    {
+        TBOX_ASSERT(d_time_stepping_type == MIDPOINT_RULE);
+        const std::string d_marker_kernel = "PIECEWISE_LINEAR";
+        d_marker_points->midpointStep(new_time - current_time, d_u_half_idx, u_new_idx, d_marker_kernel);
```

Comment:
This is set up to save the IB-ghosted data (we'll have to redo the ghost updates in the case of the scratch hierarchy ourselves here)  used to advance the structures with the midpoint rule so that we do exactly the same timestepping on the marker points.